api/src/main/java/io/grpc/InternalMetadata.java

```diff
@@ -82,4 +82,19 @@ public static Metadata newMetadata(int usedNames, byte[]... binaryValues) {
   public static int headerCount(Metadata md) {
     return md.headerCount();
   }
+
+  @Internal
+  public static Object[] serializePartial(Metadata md) {
+    return md.serializePartial();
+  }
+
+  @Internal
+  public static <T> Object parsedValue(Key<T> key, T value) {
+    return Metadata.LazyValue.create(key, value);
+  }
+
+  @Internal
+  public static Metadata newMetadataWithParsedValues(int usedNames, Object[] namesAndValues) {
```

Comment:
Note that if the run time type of the passed array is `byte[]`, it will invoke the varargs constructor 
```java
Metadata(int usedNames, byte[]... binaryValues)
```

and fail.